File: programmers/C/divisor.cpp
//정수 n을 입력받아 n의 약수를 모두 더한 값을 리턴하는 함수, solution을 완성해주세요.
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int n) {
    int answer = 0;
    for(int i=1;i<=n;i++) //약수의 합이므로 n을 포함해야 한다!
    {
        if(n%i==0) answer+=i; //약수는 나눴을 때 나머지가 0이다!
    }
    return answer;
}

File: programmers/javascript/divisor.js
//filter
function solution(arr, divisor) {
    var answer = arr.filter(v=>v%divisor == 0);
    return answer.length == 0 ? [-1] : answer.sort((a,b)=>a-b);
    
}

//map
function solution(arr, divisor) {
    var answer = [];
    arr.map((v)=> {
        v%divisor === 0 && answer.push(v);
    })
    return answer.length ? answer.sort((a,b)=>a-b) : [-1];
}

//before
function solution(arr, divisor) {
    var answer = [];

    for(var i=0; i<arr.length; i++){
        if(arr[i]%divisor == 0){
            answer.push(arr[i]);
        }
    }
    if(answer.length == 0) answer.push(-1);
    answer.sort(function(a,b){return a-b});

    return answer;
}